### ISSUE TITLE
Simulation support: to modify a model parameter should result in a solver being reinitialised

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#1549]
Say that you run the Noble 1962 model that is shipped with OpenCOR using the `Forward Euler` solver with a time step of `0.01 ms`. You could then pause the simulation, set `membrane.V` to `-20 mV`, resume the simulation and get something like:

![screen shot 2018-03-01 at 17 48 22](https://user-images.githubusercontent.com/602265/36827502-de4a8798-1d78-11e8-86fc-4784f8717a11.png)

Now, if you do something similar with the `CVODES` solver, you won't get the spike:

![screen shot 2018-03-01 at 17 49 09](https://user-images.githubusercontent.com/602265/36827510-ecddd35a-1d78-11e8-8059-5d562245cc96.png)

That's because no call to `CVodeReInit()` is made while one should have been made.

In other words, everything works as expected except with the `CVODES` solver.
